webdriver/tests/bidi/script/call_function/exception_details.py

```diff
@@ -0,0 +1,217 @@
+            await_promise=False,
+            target=ContextTarget(top_context["context"]),
+        )
+    recursive_compare(
+        {
+            "realm": any_string,
+            "exceptionDetails": {
+                "columnNumber": any_int,
+                "exception": {"type": "error"},
+                "lineNumber": any_int,
+                "stackTrace": any_stack_trace,
+                "text": any_string,
+            },
+        },
+        exception.value.result,
+    )
+
+
+@pytest.mark.asyncio
+async def test_reject_complex_value(bidi_session, top_context):
+    with pytest.raises(ScriptEvaluateResultException) as exception:
+        await bidi_session.script.call_function(
+            function_declaration="async ()=>{throw new Set(['SOME_REJECTED_RESULT'])}",
+            await_promise=True,
+            target=ContextTarget(top_context["context"]),
+        )
+
+    recursive_compare(
+        {
+            "realm": any_string,
+            "exceptionDetails": {
+                "columnNumber": any_int,
+                "exception": {
+                    "type": "set",
+                    "value": [{"type": "string", "value": "SOME_REJECTED_RESULT"}],
+                },
+                "lineNumber": any_int,
+                "stackTrace": any_stack_trace,
+                "text": any_string,
+            },
+        },
+        exception.value.result,
+    )
+
+
+@pytest.mark.asyncio
+async def test_reject_error(bidi_session, top_context):
+    with pytest.raises(ScriptEvaluateResultException) as exception:
+        await bidi_session.script.call_function(
+            function_declaration="async ()=>{throw Error('SOME_ERROR_MESSAGE')}",
+            await_promise=True,
+            target=ContextTarget(top_context["context"]),
+        )
+
+    recursive_compare(
+        {
+            "realm": any_string,
+            "exceptionDetails": {
+                "columnNumber": any_int,
+                "exception": {"type": "error"},
+                "lineNumber": any_int,
+                "stackTrace": any_stack_trace,
+                "text": any_string,
+            },
+        },
+        exception.value.result,
+    )
+
+
+@pytest.mark.asyncio
+@pytest.mark.parametrize(
+    "expression, expected",
+    [
+        ("undefined", {"type": "undefined"}),
+        ("null", {"type": "null"}),
+        ("'foobar'", {"type": "string", "value": "foobar"}),
+        ("'2'", {"type": "string", "value": "2"}),
+        ("Number.NaN", {"type": "number", "value": "NaN"}),
+        ("-0", {"type": "number", "value": "-0"}),
+        ("Infinity", {"type": "number", "value": "Infinity"}),
+        ("-Infinity", {"type": "number", "value": "-Infinity"}),
+        ("3", {"type": "number", "value": 3}),
+        ("1.4", {"type": "number", "value": 1.4}),
+        ("true", {"type": "boolean", "value": True}),
+        ("false", {"type": "boolean", "value": False}),
+        ("42n", {"type": "bigint", "value": "42"}),
```

Comment:
Why do we check all the primitives types, but only one non-primitive type?

Reply:
No strong reason, I can add a few other collections and complex types. I'm not sure how much we want to check all types all the time. Primitives are easy to cover and don't clutter the test too much.